modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,432 @@
         OptBool.new('ENUM_RVL', [ true, 'Reverse lookup a range of IP addresses', false]),
-        OptBool.new('ENUM_SRV', [ true, 'Enumerate the most common SRV records', true]),
-        OptPath.new('WORDLIST', [ false, "Wordlist for domain name bruteforcing", ::File.join(Msf::Config.data_directory, "wordlists", "namelist.txt")]),
-        OptAddress.new('NS', [ false, "Specify the nameserver to use for queries (default is system DNS)" ]),
+        OptBool.new('ENUM_TLD', [true, 'Perform a TLD expansion by replacing the TLD with the IANA TLD list', false]),
+        OptBool.new('ENUM_SRV', [true, 'Enumerate the most common SRV records', false]),
+        OptBool.new('STOP_WLDCRD', [true, 'Stops bruteforce enumeration if wildcard resolution is detected', false]),
+        OptBool.new('STOP_STORE_LOOT', [true, 'Enable Default STORE_LOOT feature in metasploit', false]),
+        OptAddress.new('NS', [false, 'Specify the nameserver to use for queries (default is system DNS)']),
         OptAddressRange.new('IPRANGE', [false, "The target address range or CIDR identifier"]),
-        OptBool.new('STOP_WLDCRD', [ true, 'Stops bruteforce enumeration if wildcard resolution is detected', false])
+        OptInt.new('THREADS', [false, 'Threads for ENUM_BRT', 1]),
+        OptPath.new('WORDLIST', [false, 'Wordlist of subdomains', ::File.join(Msf::Config.data_directory, 'wordlists', 'namelist.txt')])
       ], self.class)
 
     register_advanced_options(
       [
-        OptInt.new('RETRY', [ false, "Number of times to try to resolve a record if no response is received", 2]),
-        OptInt.new('RETRY_INTERVAL', [ false, "Number of seconds to wait before doing a retry", 2]),
-        OptBool.new('TCP_DNS', [false, "Run queries over TCP", false]),
+        OptInt.new('TIMEOUT', [false, 'DNS TIMEOUT', 8]),
+        OptInt.new('RETRY', [false, 'Number of times to try to resolve a record if no response is received', 2]),
+        OptInt.new('RETRY_INTERVAL', [false, 'Number of seconds to wait before doing a retry', 2]),
+        OptBool.new('TCP_DNS', [false, 'Run queries over TCP', false])
       ], self.class)
   end
 
-  def switchdns(target)
-    if not datastore['NS'].nil?
-      print_status("Using DNS Server: #{datastore['NS']}")
-      @res.nameserver=(datastore['NS'])
-      @nsinuse = datastore['NS']
-    else
-      querysoa = @res.query(target, "SOA")
-      if (querysoa)
-        (querysoa.answer.select { |i| i.class == Net::DNS::RR::SOA}).each do |rr|
-          query1soa = @res.search(rr.mname)
-          if (query1soa and query1soa.answer[0])
-            print_status("Setting DNS Server to #{target} NS: #{query1soa.answer[0].address}")
-            @res.nameserver=(query1soa.answer[0].address)
-            @nsinuse = query1soa.answer[0].address
-          end
-        end
-      end
-    end
-  end
+  def run
+    domain = datastore['DOMAIN']
+    return if domain.blank?
```

Comment:
Is this check actually necessary?  Can the domain ever be blank at this point?  I don't think the options would allow that.  If I am correct, then just remove these lines. If it can be blank, then put these in a `setup` method and call `fail_with` when it is blank.  That way the module won't just silently fail.